packages/name-controller/src/NameController.ts

```diff
@@ -612,4 +614,34 @@ export class NameController extends BaseController<
       delete proposedNames[dormantSourceId];
     }
   }
+
+  #cleanupExpiredEntries(): void {
+    const currentTime = this.#getCurrentTimeSeconds();
+
+    this.update((state: NameControllerState) => {
+      for (const type of Object.keys(state.names)) {
+        const nameType = type as NameType;
+        for (const value of Object.keys(state.names[nameType])) {
+          for (const variation of Object.keys(state.names[nameType][value])) {
```

Comment:
![image](https://github.com/MetaMask/core/assets/19745142/2cf09a22-182f-468c-a555-ff7540beb13c)

Reply:
Purely for readability / aesthetics, is it worth creating a `getEntriesList` that does all the looping and just returns a flat array of every `NameEntry`?

Then in this method where the logic is, we don't have to indent it 3 times so it's a little more readable, plus we can process all the entries in other ways if needed in future.

Reply:
I liked the idea, this is added here https://github.com/MetaMask/core/pull/3748/commits/2d4c932e930726b99530cb5457f3ad598039d865

Reply:
Just for reference, I think this https://github.com/MetaMask/core/pull/3748/commits/582a4fea5c3f3c74ec1d0950311d8565eecd63a3 is the commit where you added `getEntriesList`. Really amazing :1st_place_medal:  I have to say. :clap: